packages/desktop-client/src/components/reports/graphs/CalendarGraph.tsx

```diff
@@ -0,0 +1,314 @@
+              style={{
+                ...styles.tooltip,
+                lineHeight: 1.5,
+                padding: '6px 10px',
+              }}
+            >
+              <DayButton
+                key={day.date.getTime()}
+                resizeRef={index === 15 ? buttonRef : null}
+                fontSize={fontSize}
+                day={day}
+                onPress={() => onDayClick(day.date)}
+              />
+            </Tooltip>
+          ) : (
+            <DayButton
+              key={day.date.getTime()}
+              resizeRef={index === 15 ? buttonRef : null}
+              fontSize={fontSize}
+              day={day}
+              onPress={() => onDayClick(day.date)}
+            />
+          ),
+        )}
+      </View>
+    </>
+  );
+}
+
+type DayButtonProps = {
+  fontSize: number;
+  resizeRef: Ref<HTMLButtonElement>;
+  day: {
+    date: Date;
+    incomeSize: number;
+    expenseSize: number;
+  };
+  onPress: () => void;
+};
+function DayButton({ day, onPress, fontSize, resizeRef }: DayButtonProps) {
+  const [currentFontSize, setCurrentFontSize] = useState(fontSize);
+
+  useEffect(() => {
+    setCurrentFontSize(fontSize);
+  }, [fontSize]);
+
+  return (
+    <Button
+      ref={resizeRef}
+      aria-label={format(day.date, 'MMMM d, yyyy')}
+      style={{
+        borderColor: 'transparent',
+        backgroundColor: theme.calendarCellBackground,
+        position: 'relative',
+        padding: 'unset',
+        height: '100%',
+        minWidth: 0,
+        minHeight: 0,
+        margin: 0,
+      }}
+      onPress={() => onPress()}
+    >
+      {day.expenseSize !== 0 && (
+        <View
+          style={{
+            position: 'absolute',
+            width: '50%',
+            height: '100%',
+            background: chartTheme.colors.red,
+            opacity: 0.2,
+            right: 0,
+          }}
+        />
+      )}
+      {day.incomeSize !== 0 && (
+        <View
+          style={{
+            position: 'absolute',
+            width: '50%',
+            height: '100%',
+            background: chartTheme.colors.blue,
+            opacity: 0.2,
+            left: 0,
+          }}
+        />
+      )}
+      <View
+        className="bar positive-bar"
+        style={{
+          position: 'absolute',
+          left: 0,
+          bottom: 0,
+          opacity: 0.9,
+          height: `${Math.ceil(day.incomeSize)}%`,
+          backgroundColor: chartTheme.colors.blue,
+          width: '50%',
+          transition: 'height 0.5s ease-out',
+        }}
+      />
+
+      <View
+        className="bar"
+        style={{
+          position: 'absolute',
+          right: 0,
+          bottom: 0,
+          opacity: 0.9,
+          height: `${Math.ceil(day.expenseSize)}%`,
+          backgroundColor: chartTheme.colors.red,
+          width: '50%',
+          transition: 'height 0.5s ease-out',
+        }}
+      />
+      <span
+        style={{
+          fontSize: `${currentFontSize}px`,
```

Comment:
Would `AutoTextSize` component work here?

Reply:
![image](https://github.com/user-attachments/assets/00356b4a-a84e-4feb-b911-bea50920376f)
it would need a rework on the rest of the component to make it fit. its very big
but, after adjusting it each number will get its own size making it inconsistent sizing